trustar/typescript-sdk
# SDK Authentication

**What**: Implementing `TruStar` and `ApiClient` main methods for the SDK to be able to authenticate with TruSTARs API. TruStar is the main class that will forward all requests to ApiClient (maybe directly or maybe using a manager to handle logic with models. Managers are not implemented yet. Check `TruStar` constructor comments).

**Why**: These classes/methods are needed to bootstrap the SDK and be able to add more methods in the future. Our next objective is to use the JS generated files on ServiceNow.

**How**: Hitting `oauth/token` endpoint to retrieve token and using that token on header for future requests. At the moment the only other endpoint being hit is the `ping` endpoint to test that the SDK is working ok. 

Jira: https://trustar.atlassian.net/browse/EN-5145

To be able to test it, in your terminal:
 * Clone repo and checkout `feat/authentication-methods/EN-5145` branch.
 * Run `yarn install` in project's root.
 * Generate a `trustar.config.json` file containing

```json
{
  "user_api_key": <API_KEY>,
  "user_api_secret": <API_SECRET>
}
```
 * Run `tsc`
 * Run `node trustar/trusar.js`

You should see printed on the console a `pong`.
